pkg/types/reference.go

```diff
@@ -42,8 +43,8 @@ func (g *Builder) generateReferenceFields(t *types.TypeName, f *types.Var, r con
 		sfn = f.Name() + "Selector"
 	}
 
-	rn := NewNameFromCamel(rfn)
-	sn := NewNameFromCamel(sfn)
+	rn := name.NewNameFromCamel(rfn)
```

Comment:
Do we really need to move those into a new package? If yes, may be we should consider removing package name from the methods, e.g. `name.NewNameFromCamel` => name.NewFromCamel

Reply:
I had to move the name package to get rid of an import cycle between the config and types packages. Done the renaming.